### ISSUE TITLE
Issue with matplotlib dependancies

### ISSUE THREAD

**Issue by mcgheev19** [rationalmatter/Juno-Issue-Tracker#403]
**Describe the bug**

When using plt.savefig(sample+'dataworkup.pdf') I get an error that reads ModuleNotFoundError: No module named 'matplotlib._ttconv'

(where sample name is a string set at the beginning of a notebook)

**To Reproduce**
It happens for specific figures in my data workup code but some of them have no issues, but I cannot figure out what is coded differently between the cells

**Expected behavior**
Figure fails to save as PDF, but plt.show() works
**Screenshots**
<img width="710" alt="Screenshot 2024-05-22 at 2 20 11 PM" src="https://github.com/rationalmatter/Juno-Issue-Tracker/assets/32491139/fcd80d36-0215-4326-8a2f-5611fd9054d6">
<img width="657" alt="Screenshot 2024-05-22 at 2 20 32 PM" src="https://github.com/rationalmatter/Juno-Issue-Tracker/assets/32491139/bf318090-f6c5-44cd-9fda-81ff28b61266">
<img width="613" alt="Screenshot 2024-05-22 at 2 20 46 PM" src="https://github.com/rationalmatter/Juno-Issue-Tracker/assets/32491139/37bd0860-84d0-4095-9b00-9807cb3ea550">

**Environment (please complete the following information):**
 - Device: iPad Pro 11" 1st gen
 - iOS version: iOS 17.5
 - App version 3.1.0

**Comment by mcgheev19** [comment]
Update: After some more testing, this issue seems to only pop up when using plt.semilogX/Y and plt.loglog

Edit to add: I can reproducible get the error when generating log plots and trying to save as a .pdf

**Comment by alexstaravoitau** [comment]
@mcgheev19 Apologies for a late reply—this doesn't look like a screenshot of Juno though. Do you get the same error in the Juno app?

**Comment by mcgheev19** [comment]
I had saved the error output in the .ipy and then took the screenshot from my desktop, but the error was given in the Juno app.  I do not get the error if I rerun this cell from my deskstop, just in the Juno app

**Comment by alexstaravoitau** [comment]
Makes sense @mcgheev19—just double-checking to make sure the error happens in Juno too. I suspect this is because of the PDF export, which is likely missing something for rendering... I'll need to take a look, thanks for reporting.

**Comment by alexstaravoitau** [comment]
Juno 4 is now live on App Store with a full Python 3.13 env upgrade, including Matplotlib—please reopen if the issue is still present.
